sale_order_line_not_billable/models/sale_order.py

```diff
@@ -0,0 +1,15 @@
+# Copyright <2022> <Janik von Rotz - Mint System>
+# Copyright <2024> <Denis Leemann - Camptocamp SA>
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _get_invoiceable_lines(self, final=False):
+        # Filter lines with non billable products or that are sections or notes
+        invoiceable_lines = super()._get_invoiceable_lines(final=final)
+        return invoiceable_lines.filtered(
+            lambda line: line.product_id.billable
+            or line.display_type in ["line_section", "line_note"]
```

Comment:
```suggestion
            or not line.product_id
```

This should add a more general approach than just restricting to display_type (when there is a display type, there is no product)